src/api/model/modelApi.ts

```diff
@@ -14,57 +14,91 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
+import _ from 'lodash';
 
-export class ModelApi extends TransactionApi {
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
   async installModels(database: string, engine: string, models: Model[]) {
-    const action: InstallAction = {
-      type: 'InstallAction',
-      sources: models,
-    };
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """ ${model.value} """`;
+    });
 
-    return await this.runActions(database, engine, [action], false);
+    return await this.exec(database, engine, queries.join('\n'), [], false);
+  }
+
+  async installModelsAsync(database: string, engine: string, models: Model[]) {
```

Comment:
I think it also should have

```
def insert:rel:catalog:model["${model.name}"] = rel:catalog:model["${model.name}"]
```

in the query. Otherwise, you can't re-install a model.

Reply:
I'm sorry, I meant `def delete:rel:...`